modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/servlet/filter/ContentSecurityPolicyFilter.java

```diff
@@ -80,12 +90,67 @@ protected void processFilter(
 			contentSecurityPolicyConfiguration.policy();
 
 		if (Validator.isNotNull(contentSecurityPolicy)) {
+			PrintWriter httpServletResponseWriter =
+				httpServletResponse.getWriter();
+
+			ContentSecurityPolicyNonceIncludeServletResponse
+				contentSecurityPolicyNonceIncludeServletResponse =
+					new ContentSecurityPolicyNonceIncludeServletResponse(
+						httpServletResponse);
+
+			filterChain.doFilter(
+				httpServletRequest,
+				contentSecurityPolicyNonceIncludeServletResponse);
+
+			String httpServletResponseBody =
+				contentSecurityPolicyNonceIncludeServletResponse.
+					getResponseBodyAsString();
+
+			String nonce = _generateNonce();
+
+			httpServletResponseBody = httpServletResponseBody.replaceAll(
+				"<(?i)link ", "<link nonce='" + nonce + "' ");
+			httpServletResponseBody = httpServletResponseBody.replaceAll(
+				"<(?i)link>", "<link nonce='" + nonce + "'>");
+			httpServletResponseBody = httpServletResponseBody.replaceAll(
+				"<(?i)script ", "<script nonce='" + nonce + "' ");
+			httpServletResponseBody = httpServletResponseBody.replaceAll(
+				"<(?i)script>", "<script nonce='" + nonce + "'>");
+			httpServletResponseBody = httpServletResponseBody.replaceAll(
+				"<(?i)style ", "<style nonce='" + nonce + "' ");
+			httpServletResponseBody = httpServletResponseBody.replaceAll(
+				"<(?i)style>", "<style nonce='" + nonce + "'>");
```

Comment:
I think we should use double quote characters for the nonce attribute. Judging from the HTML page source of arbitrary portal pages. Or possibly make it responsive to the single vs. double quote characters being used for the tag's attribute, because both are fine according to HTML specs. Did you look into this and formed an opinion?

Reply:
You're absolutely right, I didn't think about it. Do you think we should make it configurable or follow our existing pattern with double quotes?

Reply:
I think we can go with our existing pattern of double quotes for consistency. It isn't a big issue either way so this will be enough for now.